model-mesh/crd/bases/serving.kserve.io_predictors.yaml

```diff
@@ -0,0 +1,231 @@
+                failedCopies:
+                  default: 0
+                  description:
+                    How many copies of this predictor's models failed to
+                    load recently
+                  type: integer
+                grpcEndpoint:
+                  type: string
+                httpEndpoint:
+                  description:
+                    Addressable endpoint for the deployed trained model This
+                    will be "static" and will not change when the model is mutated
+                  type: string
+                lastFailureInfo:
+                  description:
+                    Details of last failure, when load of target model is
+                    failed or blocked
+                  properties:
+                    location:
+                      description:
+                        Name of component to which the failure relates (usually
+                        Pod name)
+                      type: string
+                    message:
+                      description: Detailed error message
+                      type: string
+                    modelId:
+                      description: Internal ID of model, tied to specific Spec contents
+                      type: string
+                    reason:
+                      description: High level class of failure
+                      enum:
+                        - ModelLoadFailed
+                        - RuntimeUnhealthy
+                        - NoSupportingRuntime
+                        - RuntimeNotRecognized
+                        - InvalidPredictorSpec
+                      type: string
+                    time:
+                      description: Time failure occurred or was discovered
+                      format: date-time
+                      type: string
+                  type: object
+                targetModelState:
+                  default: ""
+                  description: ModelState enum
+                  enum:
+                    - ""
+                    - Pending
+                    - Standby
+                    - Loading
+                    - Loaded
+                    - FailedToLoad
+                  type: string
+                transitionStatus:
+                  default: UpToDate
+                  description:
+                    Whether the available predictor endpoint reflects the
+                    current Spec or is in transition
+                  enum:
+                    - UpToDate
+                    - InProgress
+                    - BlockedByFailedLoad
+                    - InvalidSpec
+                  type: string
+              required:
+                - activeModelState
+                - available
+                - failedCopies
+                - targetModelState
+                - transitionStatus
+              type: object
+          type: object
+      served: true
+      storage: true
+      subresources:
+        status: {}
+status:
```

Comment:
Aren't these lines only added once an object is created on cluster? As opposed to being needed prior to creating?